src/app/core/components/widgets/widgetcpu/widgetcpu.component.ts

```diff
@@ -144,18 +141,33 @@ export class WidgetCpuComponent extends WidgetComponent implements AfterViewInit
 
     // Calculate number of cores...
     let keys = Object.keys(data);
+    const threads = keys.filter((n) => !isNaN(parseFloat(n)));
+    let temperatureValues = [];
+
+    this.hyperthread = threads.length !== Object.keys(data.temperature).length;
 
     if(!this.coreCount){
-      this.coreCount = data.temperature ? keys.length - 2 : keys.length - 1;
+      this.threadCount = threads.length;
+      this.coreCount = this.hyperthread ? threads.length / 2 : threads.length;
     }
     
-    for(let i = 0; i < this.coreCount; i++){
+    for(let i = 0; i < this.threadCount; i++){
       usageColumn.push( parseInt(data[i.toString()].usage.toFixed(1)) );
-      if(data.temperature && data.temperature[i]){
-        temperatureColumn.push(parseInt(((data.temperature[i] / 10) - 273.05).toFixed(1)));
+
+      const mod = threads.length % 2;
+      let temperatureIndex = this.hyperthread ? Math.floor(i/2 - mod) : i;
+
+      if(data.temperature && data.temperature[temperatureIndex]){
+        temperatureValues.push(
+          parseInt(
+            (data.temperature[temperatureIndex] / 10 - 273.05).toFixed(1)
```

Comment:
Do you know what 273.05 is? Can we put this in a variable?

Reply:
Backend provides temperatures as deci-kelvin which we convert to celsius on line 163. I'll put it into a variable.